### ISSUE TITLE
Revert restrictions on gui.OpenURL

### ISSUE THREAD

**Issue by Xytime** [Facepunch/garrysmod-requests#997]
The recent restrictions on gui.OpenURL (which make it so that it only works with Steam Groups) has broken functionality in many addons. Examples of things that are broken by this: Custom commands to open ANY sort of link for a player (!forums !gametracker !donate and more), Opening links to an addon's website, so that a user can get the location to download an update, report bugs, etc. While I do understand the reason for such restrictions to be added, it doesn't really make sense to not punish the people releasing malicious addons and misusing this function on their servers, while disabling the many legit uses for this (as I said, custom commands, opening a bug tracker, etc).

Sorry if I put this in the wrong place.

**Comment by thegrb93** [comment]
Set the url to your server's steam group and put the url to your external sites there.

**Comment by neico** [comment]
Generally such restrictions (I'm referring to not only this one but also stuff like RunConsoleCommand etc.) could be handled more gracefully if they would do one of the following:

- keep the current way of restricting, but provide lua module with new unrestricted function names that each client / server has to install themself if they want to expose themself to such a risk willingly.
- modify the way of the restrictions so that they have to go through a user pop message which explains what's happening and if the user wants to expose themself to that (advantage of this is no extra installations, ex. the server could use a config file with explicitly allowed functions)

In summary, leave it up to the user, this makes the more advanced users happy and keeps malicious attempts at bay as long as the user isn't forfeiting his safety by accepting it without thinking... (same goes for windows UAC and is imo working perfectly fine)

Don't try to add "parenting control" needlessly, many users aren't children and will try to find a way to workaround a restriction which blocks whatever they want to do, which might lead to even more dangerous stuff in the end... (think closed-sourced paid modules)

**Comment by Noiwex** [comment]
I agree, the fix is lazy and harmful. It killed legitimate use completely. You killed it just straight away with no consideration of what happens next.

**Comment by Vinzuerio** [comment]
This actually breaks some Server features as well anyway
